### PR TITLE
Add default dtype for empty series

### LINKED ISSUE #165: Empty Series may change type from float64 to object
Currently we receive this warning when creating an empty pandas Series with no `dtype` specified. Happens a handful of times in `operations.py` and maybe elsewhere.

Could solve the issue by:
- Capturing the `dtype` of an empty pd.Series() somewhere (silence warning here)
- Using the `dtype=...` whenever there are no results returned

### PR DESCRIPTION
**Pull request by mesejo**
In case the content of the series if empty, the code should create the series with a the matching dtype of pandas.

Closes #165

### CODE REVIEW

File: eland/operations.py [hunk 1]
```diff
@@ -340,7 +344,8 @@ def _metric_aggs(
         # Return single value if this is a series
         # if len(numeric_source_fields) == 1:
         #    return np.float64(results[numeric_source_fields[0]])
-        s = pd.Series(data=results, index=results.keys())
+        out_dtype = EMPTY_SERIES_DTYPE if not results else None
```
Comment (sethmlarson) [inline eland/operations.py:347]: Should probably create a helper function for creating a Series (I'll be using it more often with a local change I have).

That way it's tougher to forget to do this anywhere that series are being created.

Can use a signature of ...(data,  dtype=None, **kwargs) and replace dtype if data is empty

**Comment by sethmlarson** [comment]
Jenkins test this please

**Comment by sethmlarson** [comment]
Ignore the CI failure, it's unrelated.

**Comment by sethmlarson** [comment]
Jenkins test this please

File: eland/tests/dataframe/test_dtypes_pytest.py [hunk 2]
```diff
@@ -42,3 +43,7 @@ def test_flights_select_dtypes(self):
             pd_flights.select_dtypes(include=np.number),
             ed_flights.select_dtypes(include=np.number),
         )
+
+    def test_emtpy_series_dtypes(self):
+        s = build_series({})
```
Comment (sethmlarson) [inline eland/tests/dataframe/test_dtypes_pytest.py:48]: Let's also verify that no warnings are raised. Can use record=True on capture_warnings().

File: eland/operations.py [hunk 3]
```diff
@@ -40,6 +40,16 @@
     SizeTask,
 )
 
+with warnings.catch_warnings() as w:
```
Comment (sethmlarson) [inline eland/operations.py:43]: You can remove the `w` as its not used

Comment (mesejo) [reply eland/operations.py]: Done!

Comment (sethmlarson) [reply eland/operations.py]: Oh I meant adding another capture_warning() within the test case and then doing `assert w == []` to show that your fix is working and not throwing warnings

Comment (mesejo) [reply eland/operations.py]: Done! Sorry for the mistake.

Comment (sethmlarson) [reply eland/operations.py]: No problem! I wasn't clear enough :)

**Comment by sethmlarson** [comment]
Jenkins test this please

**Comment by sethmlarson** [comment]
Thanks much :)

**Pull request merged**